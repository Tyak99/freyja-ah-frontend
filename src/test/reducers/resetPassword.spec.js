import reducer, { initialState } from '../../../store/reducers/Auth';


describe('reset password reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle LOADING', () => {
    expect(
      reducer(initialState,
        {
          type: 'LOADING',
        }),
    ).toMatchSnapshot();
  });
  it('should handle PASSWORD_RESET_ERROR', () => {
    expect(
      reducer(initialState,
        {
          type: 'PASSWORD_RESET_ERROR',
        }),
    ).toMatchSnapshot();
  });
  it('should handle PASSWORD_RESET_SUCCESS', () => {
    expect(
      reducer(initialState,
        {
          type: 'PASSWORD_RESET_SUCCESS',
        }),
    ).toMatchSnapshot();
  });
});
