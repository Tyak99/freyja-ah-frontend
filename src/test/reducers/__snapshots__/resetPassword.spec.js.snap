// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset password reducer should handle LOADING 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "isLoading": true,
  "passwordResetError": "",
  "passwordResetSuccess": "",
  "user": Object {},
}
`;

exports[`reset password reducer should handle PASSWORD_RESET_ERROR 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "isLoading": false,
  "passwordResetError": "invalid email",
  "passwordResetSuccess": "",
  "user": Object {},
}
`;

exports[`reset password reducer should handle PASSWORD_RESET_SUCCESS 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "isLoading": false,
  "passwordResetError": "",
  "passwordResetSuccess": "Please check your email",
  "user": Object {},
}
`;

exports[`reset password reducer should return the initial state 1`] = `
Object {
  "errors": Object {},
  "isAuthenticated": false,
  "isLoading": false,
  "passwordResetError": "",
  "passwordResetSuccess": "",
  "user": Object {},
}
`;
